lib/user.rb

```diff
@@ -0,0 +1,22 @@
+class User
+
+  attr_reader :name, :jokes
+
+  def initialize(name)
+    @name = name
+    @jokes = []
+  end
+
+  def learn(joke)
+    @jokes << joke
+  end
+
+  def tell(other_user, joke)
+    other_user.jokes << joke
+  end
+
+  def joke_by_id(id_num)
+    index = id_num - 1
```

Comment:
Again, this implementation is a bit brittle. Here there is a chance to use an enumerable to find an id match in the jokes array that corresponds to the argument that is passed in. Can you think of the correct enumerable?